universal-login-react/test/core/services/WalletCreationService.ts

```diff
@@ -0,0 +1,62 @@
+import {expect} from 'chai';
+import {WalletCreationService} from '../../../src/core/services/WalletCreationService';
+import UniversalLoginSDK, {WalletService} from '@universal-login/sdk';
+import {setupSdk} from '@universal-login/sdk/testutils';
+import {getWallets, createMockProvider} from 'ethereum-waffle';
+import {RelayerUnderTest} from '@universal-login/relayer';
+import {Wallet, utils} from 'ethers';
+import {INITIAL_GAS_PARAMETERS, ETHER_NATIVE_TOKEN} from '@universal-login/commons';
+
+describe('INT: WalletCreationService', () => {
+  let walletService: WalletService;
+  let sdk: UniversalLoginSDK;
+  let walletCreationService: WalletCreationService;
+  let relayer: RelayerUnderTest;
+  let wallet: Wallet;
+  let otherWallet: Wallet;
+
+  beforeEach(async () => {
+    [wallet, otherWallet] = getWallets(createMockProvider());
+    ({sdk, relayer} = await setupSdk(wallet, '34999'));
+    walletService = new WalletService(sdk);
+    walletCreationService = new WalletCreationService(walletService);
+  });
+
+  it('initiates creation flow', async () => {
+    await walletCreationService.initiateCreationFlow('name.mylogin.eth');
+    expect(walletService.state.kind).to.eq('Future');
+  });
+
+  describe('deployWhenReady', () => {
```

Comment:
What happened if we call `deployWhenReady` twice? I think that we should have a test for it. (This situation is when users cancel and go back)